tests/playwright-test/types.spec.ts

```diff
@@ -324,3 +324,77 @@ test('config should allow void/empty options', async ({ runTSC }) => {
   });
   expect(result.exitCode).toBe(0);
 });
+
+test.fixme('should check types of overloaded toMatchObject with generic parameter', async ({ runTSC }) => {
```

Comment:
why is this fixme?

Reply:
Those tests rely on stable-test-runner which doesn't have those changes yet

Reply:
These tests should resolve `@playwright/test` from the head, so should work with your change. Otherwise, all our tests would be completely wrong and testing stable-test-runner!

Reply:
I just replicated similar type tests like these:

https://github.com/microsoft/playwright/blob/0a63427c778ba447e870fe1ade72fee35c483fe1/tests/playwright-test/expect.spec.ts#L302-L308

What is the best place to put those kind of tests?

I can create some normal tests that eventually `tsc` can validate. I noticed that `lint` npm script triggers `test-types` which  ends up validating test types.

Reply:
You are right, I forgot to build playwright before running my tests. and then I got confused because I thought it was running those tests against this version:

https://github.com/microsoft/playwright/blob/main/tests/playwright-test/stable-test-runner/package.json#L4